src/main/java/org/llorllale/cactoos/matchers/EndsWith.java

```diff
@@ -52,8 +53,11 @@ public EndsWith(final Text text) {
         super(
             new TextMatcher(
                 new MatcherOf<>(
-                    (String act) -> act.endsWith(text.asString()),
-                    text
+                    act -> act.endsWith(text.asString()),
+                    desc -> desc.appendText(
+                        new FormattedText("\"%s\"", text).asString()
+                    ),
+                    (actual, desc) -> desc.appendValue(actual)
```

Comment:
@baudoliver7 let's add a construcor to `TextMatcher` based on this so that we can directly write: `new TextMatcher(act -> act.endsWith(text.asString()), "Text ending with ")`. Then let's use it everywhere it makes sense. You can either do it now or just add a todo of course :)

Reply:
There is already such a constructor :

``` java
/**
 * Ctor.
 * @param mtchr The matcher to test.
 * @param expected The description of the matcher's expected text.
 */
public TextMatcher(
    final Matcher<String> mtchr, final String expected
) {
    this(mtchr, expected, "Text is ");
}
```
And the second argument is the expected value.

Reply:
@baudoliver7 then it's not the constructor we want if it's not taking the arguments we want ;) the one I want would take a Func<String, Boolean> and an expected argument and it would delegate to MatcherOf with the whole formatted text and co.

Reply:
@victornoel I see now, with Func<String, Boolean> as first argument.

Reply:
@victornoel After having a try, it becomes more complex to introduce this constructor because we need `Matcher` expected and actual description. While `Func` doesn't have these informations, we need to add a third argument like this :
```java
/**
* Ctor.
* @param func Function to test.
* @param expected The description of the matcher's expected text.
* @param actual The description of the matcher's actual text.
*/
public TextMatcher(final Func<String, Boolean> func, final String expected, final String actual) {
  this(
      new MatcherOf<>(
          func,
          desc -> desc.appendText(
              new FormattedText("\"%s\"", expected).asString()
          ),
          (act, desc) -> desc.appendValue(act)
      ),
      actual
  );
}
```
You see here that `actual` in this constructor becomes expected argument of another constructor and `EndsWith` constructor for example from this syntax :
```java
/**
* Ctor.
* @param text The text to match against
*/
public EndsWith(final Text text) {
  super(
      new TextMatcher(
          new MatcherOf<>(
              act -> act.endsWith(text.asString()),
              desc -> desc.appendText(
                  new FormattedText("\"%s\"", text).asString()
              ),
              (actual, desc) -> desc.appendValue(actual)
          ),
          "Text ending with "
      )
  );
}
```
becomes like that :
```java
/**
* Ctor.
* @param text The text to match against
*/
public EndsWith(final Text text) throws Exception {
  super(
      new TextMatcher(
          act -> act.endsWith(text.asString()),
          text.asString(),
          "Text ending with "
      )
  );
}
```
with `throws Exception` appearing and we got in `TextMatcher` a definition of expected argument related to the context.

Reply:
@baudoliver7 what about going one step further and do this:
```java
public EndsWith(final Text text) throws Exception {
  super(
      new TextMatcher(
          txt,
          (act, txt) -> act.endsWith(txt), // or even String::endsWith
          "Text ending with "
      )
  );
}
```

and
```java
public TextMatcher(Text txt, BiFunc<String, String, Boolean> func, final String expected) {
  this(
      new MatcherOf<>(
          act -> func.apply(act, txt.asString()),
          desc -> desc.appendText(
              new FormattedText("%s \"%s\"", expected, txt).asString()
          ),
          (act, desc) -> desc.appendValue(act)
      ),
      "Text is "
  );
}
```

To be explored a bit how it works with the other matchers, but basically, let the matcher based on textmatcher be simpler to define :)

Reply:
@baudoliver7 also, I think it's strange that TextMatcher takes an expected like this and delegate to the matcher to finish the description. IMHO the fact that the expected description contains the information that the text mus end with something should be part of the wrapped matcher, not outside. So if you can simplify that also, it can be cool :)

Reply:
@victornoel I found it very strange too.

Reply:
@victornoel Good approach for your suggestion ! 👍

Reply:
@victornoel Work done :)